manager/src/test/java/com/redhat/service/smartevents/manager/ProcessorServiceConnectorTest.java

```diff
@@ -0,0 +1,328 @@
+        when(connectorsApiClient.createConnector(any(ConnectorEntity.class))).thenCallRealMethod();
+        when(connectorsApiClient.createConnector(any(ConnectorRequest.class))).thenReturn(externalConnector);
+        when(rhoasService.createTopicAndGrantAccessFor(anyString(), any())).thenReturn(new Topic());
+
+        Processor processor = processorService.createProcessor(b.getId(), b.getCustomerId(), processorRequest);
+
+        //There will be 2 re-tries at 5s each. Add 5s to be certain everything completes.
+        await().atMost(15, SECONDS).untilAsserted(() -> {
+            ConnectorEntity connector = connectorsDAO.findByProcessorIdAndName(processor.getId(),
+                    resourceNamesProvider.getProcessorConnectorName(processor.getId()));
+
+            assertThat(connector).isNotNull();
+            assertThat(connector.getError()).isNullOrEmpty();
+            assertThat(connector.getStatus()).isEqualTo(READY);
+        });
+
+        verify(rhoasService, atLeast(1)).createTopicAndGrantAccessFor(anyString(), eq(RhoasTopicAccessType.PRODUCER));
+
+        ArgumentCaptor<ConnectorRequest> connectorCaptor = ArgumentCaptor.forClass(ConnectorRequest.class);
+        verify(connectorsApiClient).createConnector(connectorCaptor.capture());
+        ConnectorRequest calledConnector = connectorCaptor.getValue();
+        assertThat(calledConnector.getKafka()).isNotNull();
+    }
+
+    @Test
+    void createConnectorFailureOnKafkaTopicCreation() {
+        Bridge b = createPersistBridge(READY);
+
+        Action slackAction = createSlackAction();
+        ProcessorRequest processorRequest = new ProcessorRequest("ManagedConnectorProcessor", slackAction);
+
+        when(rhoasService.createTopicAndGrantAccessFor(anyString(), any())).thenThrow(
+                new InternalPlatformException(RhoasServiceImpl.createFailureErrorMessageFor("errorTopic"), new RuntimeException("error")));
+        when(connectorsApiClient.createConnector(any(ConnectorRequest.class))).thenReturn(new Connector());
+
+        Processor processor = processorService.createProcessor(b.getId(), b.getCustomerId(), processorRequest);
+
+        waitForProcessorAndConnectorToFail(processor);
+
+        verify(rhoasService, atLeast(1)).createTopicAndGrantAccessFor(anyString(), eq(RhoasTopicAccessType.PRODUCER));
+        verify(connectorsApiClient, never()).createConnector(any(ConnectorRequest.class));
+    }
+
+    @Test
+    void createConnectorFailureOnExternalConnectorCreation() {
+        Bridge b = createPersistBridge(READY);
+
+        Action slackAction = createSlackAction();
+        ProcessorRequest processorRequest = new ProcessorRequest("ManagedConnectorProcessor", slackAction);
+
+        doThrow(new InternalPlatformException(RhoasServiceImpl.createFailureErrorMessageFor("errorDeletingConnector"), new RuntimeException("error")))
+                .when(connectorsApiClient).deleteConnector(anyString());
```

Comment:
Why are you mocking behaviour for `deleteConnector(..)` in a `createConnectorFailure...` test?

Reply:
I investigated and see this is just a copy and paste, so no blame on your part.

I think the test should mock this behaviour instead:
```
        doThrow(new InternalPlatformException(RhoasServiceImpl.createFailureErrorMessageFor("errorDeletingConnector"), new RuntimeException("error")))
                .when(connectorsApiClient).createConnector(any(ConnectorEntity.class));
```
I tried the modified test here and 
- The test still works as expected (i.e. passes)
- It's actually testing the right bit of code (based on the name of the test).

Reply:
I think in this refactoring we should focus on the split, then since you wrote most of that code and you feel it can be improved, you can do it in a follow-up PR. Wdyt @manstis ?

Reply:
Well, for the sake of a 1 line change I think it'd be OK adding it to this PR.

There are other changes you've made to the code-base other than splitting for this PR too.

But if you're adamant it cannot be accommodated I'll do later.